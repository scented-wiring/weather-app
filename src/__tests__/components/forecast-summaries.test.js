import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/forecast-summaries";

describe("ForecastSummaries", () => {
  const forecasts = [
    {
      date: 1525046400000,
      description: "date1",
      icon: "800",
      temperature: {
        max: 999,
      },
    },
    {
      date: 1525132800000,
      description: "date2",
      icon: "211",
      temperature: {
        max: 777,
      },
    },
  ];

  it("renders the correct amount of ForecastSummary components", () => {
    const { asFragment } = render(<ForecastSummaries forecasts={forecasts} />);

    expect(asFragment).toMatchSnapshot();
  });

  it("renders the correct amount of ForecastSummary component props", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries forecasts={forecasts} />
    );

    expect(getAllByTestId("date-id")).toHaveLength(2);
    expect(getAllByTestId("description-id")).toHaveLength(2);
    expect(getAllByTestId("icon-id")).toHaveLength(2);
    expect(getAllByTestId("temperature-id")).toHaveLength(2);
  });
});
