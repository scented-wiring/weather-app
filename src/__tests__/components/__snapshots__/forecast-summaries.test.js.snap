// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders the correct amount of ForecastSummary components 1`] = `[Function]`;
